### ISSUE TITLE
[BUG][SDW] s2idle during aplay: Load fw will fail on resume

### ISSUE THREAD

**Issue by ClarexZhou** [thesofproject/linux#1196]
**Describe the bug**
aplay stopped for input/out error. load fw failed then ipc timed out for TRIG_STOP when enter and wake from s2idle during aplay.

**To Reproduce**
1. echo s2idle > /sys/power/mem_sleep
2. Terminal 1: aplay -Dhw:0,0 -r48000 -c2 -fs16_le /dev/zero -vv -i
3. Terminal 2: sudo rtcwake -m mem -s 10

**frequency**
2/2

**Actual result**
aplay stopped for input/out error. load fw failed then ipc timed out for TRIG_STOP when enter and wake from s2idle during aplay.

**Expected result**
Wake up and aplay should be resumed normally.

**Environment**
linux: https://github.com/thesofproject/linux/tree/topic/soundwire commit: bccf4a8a
sof: https://github.com/thesofproject/sof/commits/sdw-001-drop-stable commit: d9c0463
Topology: file: sof-cml-rt700.m4 same path as firmware
platform: CML SDW RVP

No error messages.

**dmesg**
```
[  166.990456] PM: suspend entry (s2idle)
[  166.990711] Filesystems sync: 0.000 seconds
[  166.991634] Freezing user space processes ... (elapsed 0.001 seconds) done.
[  166.993525] OOM killer disabled.
[  166.993526] Freezing remaining freezable tasks ... (elapsed 0.001 seconds) done.
[  166.994751] printk: Suspending console(s) (use no_console_suspend to debug)
[  166.995497] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0x20140000 successful
[  166.995512] sof-audio-pci 0000:00:1f.3: ipc tx: 0x40010000: GLB_PM_MSG: CTX_SAVE
[  166.995820] sof-audio-pci 0000:00:1f.3: ipc tx succeeded: 0x40010000: GLB_PM_MSG: CTX_SAVE
[  166.995847] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0x1010f0f successful
[  166.995854] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0xf0f successful
[  166.995858] sof-audio-pci 0000:00:1f.3: DSP core(s) enabled? 0 : core_mask f
[  166.995922] sof-audio-pci 0000:00:1f.3: Debug PCIR: 00000010 at  00000044
[  167.299443] ACPI: EC: interrupt blocked
[  177.633050] ACPI: EC: interrupt unblocked
[  177.669328] hpet: Lost 766 RTC interrupts
[  177.712674] serial 00:06: activated
[  177.931235] [drm] Reducing the compressed framebuffer size. This may lead to less power savings than a non-reduced-size. Try to increase stolen memory size if available in BIOS.
[  177.939936] atkbd serio0: Failed to deactivate keyboard on isa0060/serio0
[  178.020747] ata3: SATA link down (SStatus 4 SControl 300)
[  178.128686] usb 1-7: reset low-speed USB device number 3 using xhci_hcd
[  178.316130] atkbd serio0: Failed to enable keyboard on isa0060/serio0
[  178.404740] sof-audio-pci 0000:00:1f.3: Debug PCIR: 00000000 at  00000044
[  178.404749] sof-audio-pci 0000:00:1f.3: Debug PCIR: 00000040 at  00000048
[  178.404755] sof-audio-pci 0000:00:1f.3: Debug PCIW: 00000000 at  00000048
[  178.408197] sof-audio-pci 0000:00:1f.3: Debug PCIR: 00000000 at  00000048
[  178.408205] sof-audio-pci 0000:00:1f.3: Debug PCIW: 00000040 at  00000048
[  178.408909] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0x1010f0f successful
[  178.408917] sof-audio-pci 0000:00:1f.3: loading firmware
[  178.408927] sof-audio-pci 0000:00:1f.3: Debug PCIR: 00000002 at  00000048
[  178.408934] sof-audio-pci 0000:00:1f.3: Debug PCIW: 00000000 at  00000048
[  178.408941] sof-audio-pci 0000:00:1f.3: Debug PCIR: 00000000 at  00000044
[  178.408947] sof-audio-pci 0000:00:1f.3: Debug PCIW: 00000004 at  00000044
[  178.408949] sof-audio-pci 0000:00:1f.3: booting DSP firmware
[  178.409023] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0x40000 successful
[  178.409046] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0x40000 successful
[  178.409052] sof-audio-pci 0000:00:1f.3: period_bytes:0x0
[  178.409054] sof-audio-pci 0000:00:1f.3: periods:1
[  178.409137] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0xf0f0f0f successful
[  178.409148] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0xf0f0f0e successful
[  178.409153] sof-audio-pci 0000:00:1f.3: unstall/run core: core_mask = 1
[  178.409160] sof-audio-pci 0000:00:1f.3: DSP core(s) enabled? 1 : core_mask 1
[  178.409772] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0x80000000 successful
[  178.409781] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0x1010e0e successful
[  178.409918] sof-audio-pci 0000:00:1f.3: error: no reply expected, received 0x0
[  178.502966] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0x5000001 successful
[  180.146321] asix 1-3:1.0 enx000ec6caa0a8: link up, 100Mbps, full-duplex, lpa 0xCDE1
[  181.503534] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0x5000001 timedout
[  181.503901] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0x240002 timedout
[  181.503904] sof-audio-pci 0000:00:1f.3: error: DMA trigger stop failed
[  181.503907] sof-audio-pci 0000:00:1f.3: error: load fw failed ret: -110
[  181.503938] sof-audio-pci 0000:00:1f.3: unknown ROM status value 05000001
[  181.503945] sof-audio-pci 0000:00:1f.3: error: status = 0x00000000 panic = 0x00000000
[  181.503951] sof-audio-pci 0000:00:1f.3: unknown ROM status value 05000001
[  181.503957] sof-audio-pci 0000:00:1f.3: error: failed to reset DSP
[  181.503960] sof-audio-pci 0000:00:1f.3: error: failed to boot DSP firmware after resume -110
[  181.503970] PM: dpm_run_callback(): pci_pm_resume+0x0/0xa0 returns -110
[  181.503979] PM: Device 0000:00:1f.3 failed to resume: error -110
[  181.504050] HDMI HDA Codec ehdaudio0D2: hdac_hdmi_present_sense: disconnect for pin:port 5:0
[  181.504055] HDMI HDA Codec ehdaudio0D2: hdac_hdmi_present_sense: disconnect for pin:port 6:0
[  181.504059] HDMI HDA Codec ehdaudio0D2: hdac_hdmi_present_sense: disconnect for pin:port 7:0
[  181.534908] int-sdw int-sdw.1: intel_resume done
[  181.536179] OOM killer enabled.
[  181.536181] Restarting tasks ... done.
[  181.586391] PM: suspend exit
[  182.748195] plb: soc_pcm_trigger codec_dai name snd-soc-dummy-dai
[  182.748205] plb: soc_pcm_trigger component name 0000:00:1f.3
[  182.748218] sof-audio-pci 0000:00:1f.3: pcm: trigger stream 0 dir 0 cmd 0
[  182.748230] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0x40000 successful
[  182.748241] sof-audio-pci 0000:00:1f.3: ipc tx: 0x60050000: GLB_STREAM_MSG: TRIG_STOP
[  183.252135] sof-audio-pci 0000:00:1f.3: error: ipc timed out for 0x60050000 size 12
[  183.252150] sof-audio-pci 0000:00:1f.3: unknown ROM status value 01009701
[  183.252157] sof-audio-pci 0000:00:1f.3: error: status = 0x001d003c panic = 0x00000000
[  183.252164] sof-audio-pci 0000:00:1f.3: unknown ROM status value 01009701
[  183.252175] sof-audio-pci 0000:00:1f.3: error: hda irq intsts 0x00000000 intlctl 0x00000000 rirb 00
[  183.252178] sof-audio-pci 0000:00:1f.3: error: dsp irq ppsts 0x00000000 adspis 0x00000000
[  183.252185] sof-audio-pci 0000:00:1f.3: error: host status 0x00000000 dsp status 0x00040001 mask 0x00000000
[  183.252191]  SDW1-speakers: ASoC: trigger FE failed -110
[  183.252220] sof-audio-pci 0000:00:1f.3: pcm: free stream 0 dir 0
[  183.252226] sof-audio-pci 0000:00:1f.3: ipc tx: 0x60030000: GLB_STREAM_MSG: PCM_FREE
[  183.756137] sof-audio-pci 0000:00:1f.3: error: ipc timed out for 0x60030000 size 12
[  183.756152] sof-audio-pci 0000:00:1f.3: unknown ROM status value 01009701
[  183.756159] sof-audio-pci 0000:00:1f.3: error: status = 0x001d003c panic = 0x00000000
[  183.756166] sof-audio-pci 0000:00:1f.3: unknown ROM status value 01009701
[  183.756177] sof-audio-pci 0000:00:1f.3: error: hda irq intsts 0x00000000 intlctl 0x00000000 rirb 00
[  183.756181] sof-audio-pci 0000:00:1f.3: error: dsp irq ppsts 0x00000000 adspis 0x00000000
[  183.756190] sof-audio-pci 0000:00:1f.3: error: host status 0x00000000 dsp status 0x00040001 mask 0x00000000
[  185.796096] int-sdw int-sdw.1: IO transfer timed out
[  185.796123] sof-audio-pci 0000:00:1f.3: ipc tx: 0x80010000: GLB_DAI_MSG: CONFIG
[  186.300106] sof-audio-pci 0000:00:1f.3: error: ipc timed out for 0x80010000 size 120
[  186.300114] sof-audio-pci 0000:00:1f.3: unknown ROM status value 01009701
[  186.300117] sof-audio-pci 0000:00:1f.3: error: status = 0x001d003c panic = 0x00000000
[  186.300120] sof-audio-pci 0000:00:1f.3: unknown ROM status value 01009701
[  186.300125] sof-audio-pci 0000:00:1f.3: error: hda irq intsts 0x00000000 intlctl 0x00000000 rirb 0
```
[s2idle_failedDuringAplay_dmesg.txt](https://github.com/thesofproject/linux/files/3603378/s2idle_failedDuringAplay_dmesg.txt)
[s2idle_failedDuringAplay_logger.txt](https://github.com/thesofproject/linux/files/3603379/s2idle_failedDuringAplay_logger.txt)

**Comment by plbossart** [comment]
suspend is not supported for now

**Comment by ClarexZhou** [comment]
Runtime PM status can enter suspended and active again normally on CML rt700 SDW RVP.
Can be tested without "options snd_sof_pci sof_debug=1" in /etc/modprobe.d/alsa-base.conf 
also MSI is disabled via "options snd_sof_intel_hda_common use_msi=0"

Error seen with firmware master and kernel integration/soundwire-latest branch.
**Environment**
Firmware: 460c41a  Master
Kernel: 4ac5440 on integration/soundwire-latest
Topology: file: same with firmware branch, sof-cml-rt700.m4
platform: CML rt700 SDW RVP

**Comment by YvonneYang2** [comment]
Same as https://github.com/thesofproject/linux/issues/1196#issuecomment-532991012. Error can also be seen on ICL RVP, with codec ALC711 -RT1308, RT715 in SDW mode 

Test recipe:
Firmware: 460c41aa https://github.com/thesofproject/sof/commits/master
Kernel: 4ac5440 https://github.com/thesofproject/linux/tree/integration/soundwire-latest
tplg:  sof-icl-rt711-rt1308-rt715-hdmi.tplg rename sof-icl-rt711-rt1308-rt715.tplg  https://github.com/bardliao/sof/blob/fix/soundwire-tplg/tools/topology/
platform: ICL RVP, with codec ALC711 -RT1308, RT715 in SDW mode

**Comment by plbossart** [comment]
suspend-resume issue, I am not even going to look at it since the code is not present in the latest branch

**Comment by mengdonglin** [comment]
Depend on feature https://github.com/thesofproject/linux/issues/1238

**Comment by YvonneYang2** [comment]
Issue can be reproduced on icl rvp rt700 when wake up from s2idle suspend during paplay 

**Test preconditions:**
make sure both options are commented out.
#options snd-sof-pci sof_debug=1
#options soundwire_intel sdw_md_flags=0x01010101

**Test recipe:**
Firmware: 9794dd7 https://github.com/thesofproject/sof/commits/master + +PR1851 (For HDMI tplg)
Kernel: 07bdd03 https://github.com/thesofproject/linux/tree/integration/soundwire-latest-pm
Topology: Same with FW branch: sof-icl-rt700.m4
Platform:ICL RVP with codec ALC700 in SDW mode

[paplay_sus_dmesg.log](https://github.com/thesofproject/linux/files/3645488/paplay_sus_dmesg.log)
[paplay_sus_logger.log](https://github.com/thesofproject/linux/files/3645489/paplay_sus_logger.log)

**Comment by ClarexZhou** [comment]
Error still seen on latest kernel
**Environment**
Firmware: e755494  Master
Kernel: d2b55e0 on integration/soundwire-latest
Topology: file: same with firmware branch, sof-cml-rt700.m4
platform: CML RVP with onboard ALC700 in SDW mode

**Comment by YvonneYang2** [comment]
Error can been seen on latest kernel and FW on icl 3-in-1 rvp 

Environment
Firmware: c7d0149 https://github.com/thesofproject/sof/commits/master 
Kernel: e7fb5a5 https://github.com/thesofproject/linux/tree/integration/soundwire-latest
tplg:  sof-icl-rt711-rt1308-rt715-hdmi.tplg rename sof-icl-rt711-rt1308-rt715.tplg  path same as FW branch
platform: ICL RVP, with codec ALC711 -RT1308, RT715 in SDW mode

**Comment by plbossart** [comment]
Please recheck with the latest integration/soundwire-latest (19df577a9096437d50ad77fa5af34f90a8f858da), these errors are gone for me.

The behavior is not fully right just yet, so added new issue https://github.com/thesofproject/linux/issues/1311

**Comment by YvonneYang2** [comment]
Test it with latest latest integration/soundwire-latest (19df577) on icl 3-in-1 rvp and cml rt700 rvp, these errors are gone. New behavior will track in #1311
